sdk/core/Azure.Core/src/Pipeline/RetryPolicyOptions.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Core.Pipeline
+{
+    /// <summary>
+    /// The set of options to use for configuring the retry policy.
+    /// </summary>
+    public class RetryPolicyOptions
```

Comment:
Would we want this to include configuration for everything in the policy, e.g. also NetworkTimeout and related from RetryOptions?

Reply:
So NetworkTimeout is interesting - it actually isn't used by the RetryPolicy, but the only way to configure it currently is via RetryOptions. It actually isn't even related to the concept of retries. However, given that we bill RetryPolicy as the replacement for RetryOptions, it does introduce a potential point of confusion. IMO, it would be better if NetworkTimeout was a top-level property on ClientOptions, and we EBN'ed the one on RetryOptions.

Reply:
Cool, that makes sense to me.  

Again to ground my understanding - if a caller sets a RetryPolicy, does that mean we completely ignore RetryOptions?  (It sounds like what you're saying above, but I want to confirm I'm understanding correctly.)  If that is correct, it might be worth saying that explicitly in the docs - the following piece may just be out of date, but it seems to contradict what we're saying here (or I'm just totally confused :))

> Notice that the policy takes in RetryOptions as one of the constructor parameters and passes it to the base constructor. By doing this, we are able to delegate to the base RetryPolicy as needed (either by explicitly invoking the base methods, or by not overriding methods that we do not need to customize) which will respect the RetryOptions.

Reply:
NetworkTimeout feels related to the transport rather than a top level client option. If we decide to do something like this, I think we should consider adding it to a `TransportOptions` on `ClientOptions` like we were going to do for `IsClientRedirectEnabled`

Reply:
> Again to ground my understanding - if a caller sets a RetryPolicy, does that mean we completely ignore RetryOptions? (It sounds like what you're saying above, but I want to confirm I'm understanding correctly.)

Nope, we still will [thread the NetworkTimeout](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs#L184) through to the ResponseBodyPolicy. I agree that the story is a bit confusing with how we are presenting RetryPolicy as an override to RetryOptions. I would suggest we consider moving NetworkTimeout somewhere that is more sensible (perhaps TransportOptions) as @christothes suggests.

Reply:
Filed https://github.com/Azure/azure-sdk-for-net/issues/37736